From d1a7522add88a55c1672de5489e5aea5eacaada6 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Thu, 3 Mar 2016 21:55:27 -0500
Subject: [PATCH] wr-policy: fix avc denial while sysvinit rebooting

Reference sources: selinux-policy-3.13.1-128.21.fc22.src.rpm
Changes come from: policy-f22-base.patch

* Allow rpcd_t write to proc_t file
* Allow alsa_t read, write /dev/console
* Allow syslogd_t read all level files
* Get kernel_t, sysadm_t create /dev/* with correct labels

Fix avc denial:

  avc: denied { write } for pid=1101 comm="sm-notify" \
  name="nlm_end_grace" dev="proc" ino=4026532077 \
  scontext=system_u:system_r:rpcd_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:proc_t:s0 \
  tclass=file permissive=0

  avc: denied { read write } for pid=997 comm="alsactl" \
  path="/dev/console" dev="devtmpfs" ino=7265 \
  scontext=system_u:system_r:alsa_t:s0-s15:c0.c1023 \
  tcontext=root:object_r:user_tty_device_t:s0 \
  tclass=chr_file permissive=0

  avc: denied { read write } for pid=68 comm="mount" \
  path="/dev/console" dev="devtmpfs" ino=7265 \
  scontext=system_u:system_r:mount_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:event_device_t:s15:c0.c1023 \
  tclass=chr_file permissive=1

  avc: denied { read } for pid=534 comm="rsyslogd" \
  name="kmsg" dev="proc" ino=4026532002 \
  scontext=system_u:system_r:syslogd_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:proc_kmsg_t:s15:c0.c1023 \
  tclass=file permissive=1

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/contrib/alsa.te    |   3 +
 policy/modules/contrib/rpc.te     |   2 +
 policy/modules/kernel/kernel.te   |   2 +
 policy/modules/kernel/terminal.if | 395 ++++++++++++++++++++++++++++++++++++++
 policy/modules/roles/sysadm.te    |   2 +
 policy/modules/system/logging.te  |   1 +
 6 files changed, 405 insertions(+)

--- refpolicy.orig/policy/modules/contrib/alsa.te
+++ refpolicy/policy/modules/contrib/alsa.te
@@ -99,6 +99,9 @@ userdom_manage_unpriv_user_semaphores(al
 userdom_manage_unpriv_user_shared_mem(alsa_t)
 userdom_search_user_home_dirs(alsa_t)
 
+term_use_all_ttys(alsa_t)
+term_use_all_ptys(alsa_t)
+
 ifdef(`distro_debian',`
 	term_dontaudit_use_unallocated_ttys(alsa_t)
 
--- refpolicy.orig/policy/modules/contrib/rpc.te
+++ refpolicy/policy/modules/contrib/rpc.te
@@ -156,6 +156,8 @@ files_pid_filetrans(rpcd_t, rpcd_var_run
 
 can_exec(rpcd_t, rpcd_exec_t)
 
+kernel_read_system_state(rpcd_t)
+kernel_write_proc_files(rpcd_t)
 kernel_read_network_state(rpcd_t)
 kernel_read_sysctl(rpcd_t)
 kernel_rw_fs_sysctls(rpcd_t)
--- refpolicy.orig/policy/modules/kernel/kernel.te
+++ refpolicy/policy/modules/kernel/kernel.te
@@ -277,6 +277,8 @@ dev_setattr_generic_chr_files(kernel_t)
 dev_getattr_fs(kernel_t)
 dev_getattr_sysfs(kernel_t)
 
+term_filetrans_all_named_dev(kernel_t)
+
 # Mount root file system. Used when loading a policy
 # from initrd, then mounting the root filesystem
 fs_mount_all_fs(kernel_t)
--- refpolicy.orig/policy/modules/kernel/terminal.if
+++ refpolicy/policy/modules/kernel/terminal.if
@@ -1626,3 +1626,398 @@ interface(`term_use_virtio_console',`
 	dev_list_all_dev_nodes($1)
 	allow $1 virtio_device_t:chr_file rw_term_perms;
 ')
+
+########################################
+## <summary>
+##      Create all named term devices with the correct label
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+#
+interface(`term_filetrans_all_named_dev',`
+        gen_require(`
+            type tty_device_t;
+            type bsdpty_device_t;
+            type console_device_t;
+            type ptmx_t;
+            type devtty_t;
+            type virtio_device_t;
+            type devpts_t;
+            type usbtty_device_t;
+        ')
+
+        dev_filetrans($1, devtty_t, chr_file, "tty")
+        dev_filetrans($1, tty_device_t, chr_file, "tty0")
+        dev_filetrans($1, tty_device_t, chr_file, "tty1")
+        dev_filetrans($1, tty_device_t, chr_file, "tty2")
+        dev_filetrans($1, tty_device_t, chr_file, "tty3")
+        dev_filetrans($1, tty_device_t, chr_file, "tty4")
+        dev_filetrans($1, tty_device_t, chr_file, "tty5")
+        dev_filetrans($1, tty_device_t, chr_file, "tty6")
+        dev_filetrans($1, tty_device_t, chr_file, "tty7")
+        dev_filetrans($1, tty_device_t, chr_file, "tty8")
+        dev_filetrans($1, tty_device_t, chr_file, "tty9")
+        dev_filetrans($1, tty_device_t, chr_file, "tty10")
+        dev_filetrans($1, tty_device_t, chr_file, "tty11")
+        dev_filetrans($1, tty_device_t, chr_file, "tty12")
+        dev_filetrans($1, tty_device_t, chr_file, "tty13")
+        dev_filetrans($1, tty_device_t, chr_file, "tty14")
+        dev_filetrans($1, tty_device_t, chr_file, "tty15")
+        dev_filetrans($1, tty_device_t, chr_file, "tty16")
+        dev_filetrans($1, tty_device_t, chr_file, "tty17")
+        dev_filetrans($1, tty_device_t, chr_file, "tty18")
+        dev_filetrans($1, tty_device_t, chr_file, "tty19")
+        dev_filetrans($1, tty_device_t, chr_file, "tty20")
+        dev_filetrans($1, tty_device_t, chr_file, "tty21")
+        dev_filetrans($1, tty_device_t, chr_file, "tty22")
+        dev_filetrans($1, tty_device_t, chr_file, "tty23")
+        dev_filetrans($1, tty_device_t, chr_file, "tty24")
+        dev_filetrans($1, tty_device_t, chr_file, "tty25")
+        dev_filetrans($1, tty_device_t, chr_file, "tty26")
+        dev_filetrans($1, tty_device_t, chr_file, "tty27")
+        dev_filetrans($1, tty_device_t, chr_file, "tty28")
+        dev_filetrans($1, tty_device_t, chr_file, "tty29")
+        dev_filetrans($1, tty_device_t, chr_file, "tty30")
+        dev_filetrans($1, tty_device_t, chr_file, "tty31")
+        dev_filetrans($1, tty_device_t, chr_file, "tty32")
+        dev_filetrans($1, tty_device_t, chr_file, "tty33")
+        dev_filetrans($1, tty_device_t, chr_file, "tty34")
+        dev_filetrans($1, tty_device_t, chr_file, "tty35")
+        dev_filetrans($1, tty_device_t, chr_file, "tty36")
+        dev_filetrans($1, tty_device_t, chr_file, "tty37")
+        dev_filetrans($1, tty_device_t, chr_file, "tty38")
+        dev_filetrans($1, tty_device_t, chr_file, "tty39")
+        dev_filetrans($1, tty_device_t, chr_file, "tty40")
+        dev_filetrans($1, tty_device_t, chr_file, "tty41")
+        dev_filetrans($1, tty_device_t, chr_file, "tty42")
+        dev_filetrans($1, tty_device_t, chr_file, "tty43")
+        dev_filetrans($1, tty_device_t, chr_file, "tty44")
+        dev_filetrans($1, tty_device_t, chr_file, "tty45")
+        dev_filetrans($1, tty_device_t, chr_file, "tty46")
+        dev_filetrans($1, tty_device_t, chr_file, "tty47")
+        dev_filetrans($1, tty_device_t, chr_file, "tty48")
+        dev_filetrans($1, tty_device_t, chr_file, "tty49")
+        dev_filetrans($1, tty_device_t, chr_file, "tty50")
+        dev_filetrans($1, tty_device_t, chr_file, "tty51")
+        dev_filetrans($1, tty_device_t, chr_file, "tty52")
+        dev_filetrans($1, tty_device_t, chr_file, "tty53")
+        dev_filetrans($1, tty_device_t, chr_file, "tty54")
+        dev_filetrans($1, tty_device_t, chr_file, "tty55")
+        dev_filetrans($1, tty_device_t, chr_file, "tty56")
+        dev_filetrans($1, tty_device_t, chr_file, "tty57")
+        dev_filetrans($1, tty_device_t, chr_file, "tty58")
+        dev_filetrans($1, tty_device_t, chr_file, "tty59")
+        dev_filetrans($1, tty_device_t, chr_file, "tty60")
+        dev_filetrans($1, tty_device_t, chr_file, "tty61")
+        dev_filetrans($1, tty_device_t, chr_file, "tty62")
+        dev_filetrans($1, tty_device_t, chr_file, "tty63")
+        dev_filetrans($1, tty_device_t, chr_file, "tty64")
+        dev_filetrans($1, tty_device_t, chr_file, "tty65")
+        dev_filetrans($1, tty_device_t, chr_file, "tty66")
+        dev_filetrans($1, tty_device_t, chr_file, "tty67")
+        dev_filetrans($1, tty_device_t, chr_file, "tty68")
+        dev_filetrans($1, tty_device_t, chr_file, "tty69")
+        dev_filetrans($1, tty_device_t, chr_file, "tty70")
+        dev_filetrans($1, tty_device_t, chr_file, "tty71")
+        dev_filetrans($1, tty_device_t, chr_file, "tty72")
+        dev_filetrans($1, tty_device_t, chr_file, "tty73")
+        dev_filetrans($1, tty_device_t, chr_file, "tty74")
+        dev_filetrans($1, tty_device_t, chr_file, "tty75")
+        dev_filetrans($1, tty_device_t, chr_file, "tty76")
+        dev_filetrans($1, tty_device_t, chr_file, "tty77")
+        dev_filetrans($1, tty_device_t, chr_file, "tty78")
+        dev_filetrans($1, tty_device_t, chr_file, "tty79")
+        dev_filetrans($1, tty_device_t, chr_file, "tty80")
+        dev_filetrans($1, tty_device_t, chr_file, "tty81")
+        dev_filetrans($1, tty_device_t, chr_file, "tty82")
+        dev_filetrans($1, tty_device_t, chr_file, "tty83")
+        dev_filetrans($1, tty_device_t, chr_file, "tty84")
+        dev_filetrans($1, tty_device_t, chr_file, "tty85")
+        dev_filetrans($1, tty_device_t, chr_file, "tty86")
+        dev_filetrans($1, tty_device_t, chr_file, "tty87")
+        dev_filetrans($1, tty_device_t, chr_file, "tty88")
+        dev_filetrans($1, tty_device_t, chr_file, "tty89")
+        dev_filetrans($1, tty_device_t, chr_file, "tty90")
+        dev_filetrans($1, tty_device_t, chr_file, "tty91")
+        dev_filetrans($1, tty_device_t, chr_file, "tty92")
+        dev_filetrans($1, tty_device_t, chr_file, "tty93")
+        dev_filetrans($1, tty_device_t, chr_file, "tty94")
+        dev_filetrans($1, tty_device_t, chr_file, "tty95")
+        dev_filetrans($1, tty_device_t, chr_file, "tty96")
+        dev_filetrans($1, tty_device_t, chr_file, "tty97")
+        dev_filetrans($1, tty_device_t, chr_file, "tty98")
+        dev_filetrans($1, tty_device_t, chr_file, "tty99")
+        dev_filetrans($1, tty_device_t, chr_file, "pty")
+        dev_filetrans($1, tty_device_t, chr_file, "pty0")
+        dev_filetrans($1, tty_device_t, chr_file, "pty1")
+        dev_filetrans($1, tty_device_t, chr_file, "pty2")
+        dev_filetrans($1, tty_device_t, chr_file, "pty3")
+        dev_filetrans($1, tty_device_t, chr_file, "pty4")
+        dev_filetrans($1, tty_device_t, chr_file, "pty5")
+        dev_filetrans($1, tty_device_t, chr_file, "pty6")
+        dev_filetrans($1, tty_device_t, chr_file, "pty7")
+        dev_filetrans($1, tty_device_t, chr_file, "pty8")
+        dev_filetrans($1, tty_device_t, chr_file, "pty9")
+        dev_filetrans($1, tty_device_t, chr_file, "pty10")
+        dev_filetrans($1, tty_device_t, chr_file, "pty11")
+        dev_filetrans($1, tty_device_t, chr_file, "pty12")
+        dev_filetrans($1, tty_device_t, chr_file, "pty13")
+        dev_filetrans($1, tty_device_t, chr_file, "pty14")
+        dev_filetrans($1, tty_device_t, chr_file, "pty15")
+        dev_filetrans($1, tty_device_t, chr_file, "pty16")
+        dev_filetrans($1, tty_device_t, chr_file, "pty17")
+        dev_filetrans($1, tty_device_t, chr_file, "pty18")
+        dev_filetrans($1, tty_device_t, chr_file, "pty19")
+        dev_filetrans($1, tty_device_t, chr_file, "pty20")
+        dev_filetrans($1, tty_device_t, chr_file, "pty21")
+        dev_filetrans($1, tty_device_t, chr_file, "pty22")
+        dev_filetrans($1, tty_device_t, chr_file, "pty23")
+        dev_filetrans($1, tty_device_t, chr_file, "pty24")
+        dev_filetrans($1, tty_device_t, chr_file, "pty25")
+        dev_filetrans($1, tty_device_t, chr_file, "pty26")
+        dev_filetrans($1, tty_device_t, chr_file, "pty27")
+        dev_filetrans($1, tty_device_t, chr_file, "pty28")
+        dev_filetrans($1, tty_device_t, chr_file, "pty29")
+        dev_filetrans($1, tty_device_t, chr_file, "pty30")
+        dev_filetrans($1, tty_device_t, chr_file, "pty31")
+        dev_filetrans($1, tty_device_t, chr_file, "pty32")
+        dev_filetrans($1, tty_device_t, chr_file, "pty33")
+        dev_filetrans($1, tty_device_t, chr_file, "pty34")
+        dev_filetrans($1, tty_device_t, chr_file, "pty35")
+        dev_filetrans($1, tty_device_t, chr_file, "pty36")
+        dev_filetrans($1, tty_device_t, chr_file, "pty37")
+        dev_filetrans($1, tty_device_t, chr_file, "pty38")
+        dev_filetrans($1, tty_device_t, chr_file, "pty39")
+        dev_filetrans($1, tty_device_t, chr_file, "pty40")
+        dev_filetrans($1, tty_device_t, chr_file, "pty41")
+        dev_filetrans($1, tty_device_t, chr_file, "pty42")
+        dev_filetrans($1, tty_device_t, chr_file, "pty43")
+        dev_filetrans($1, tty_device_t, chr_file, "pty44")
+        dev_filetrans($1, tty_device_t, chr_file, "pty45")
+        dev_filetrans($1, tty_device_t, chr_file, "pty46")
+        dev_filetrans($1, tty_device_t, chr_file, "pty47")
+        dev_filetrans($1, tty_device_t, chr_file, "pty48")
+        dev_filetrans($1, tty_device_t, chr_file, "pty49")
+        dev_filetrans($1, tty_device_t, chr_file, "pty50")
+        dev_filetrans($1, tty_device_t, chr_file, "pty51")
+        dev_filetrans($1, tty_device_t, chr_file, "pty52")
+        dev_filetrans($1, tty_device_t, chr_file, "pty53")
+        dev_filetrans($1, tty_device_t, chr_file, "pty54")
+        dev_filetrans($1, tty_device_t, chr_file, "pty55")
+        dev_filetrans($1, tty_device_t, chr_file, "pty56")
+        dev_filetrans($1, tty_device_t, chr_file, "pty57")
+        dev_filetrans($1, tty_device_t, chr_file, "pty58")
+        dev_filetrans($1, tty_device_t, chr_file, "pty59")
+        dev_filetrans($1, tty_device_t, chr_file, "pty60")
+        dev_filetrans($1, tty_device_t, chr_file, "pty61")
+        dev_filetrans($1, tty_device_t, chr_file, "pty62")
+        dev_filetrans($1, tty_device_t, chr_file, "pty63")
+        dev_filetrans($1, tty_device_t, chr_file, "pty64")
+        dev_filetrans($1, tty_device_t, chr_file, "pty65")
+        dev_filetrans($1, tty_device_t, chr_file, "pty66")
+        dev_filetrans($1, tty_device_t, chr_file, "pty67")
+        dev_filetrans($1, tty_device_t, chr_file, "pty68")
+        dev_filetrans($1, tty_device_t, chr_file, "pty69")
+        dev_filetrans($1, tty_device_t, chr_file, "pty70")
+        dev_filetrans($1, tty_device_t, chr_file, "pty71")
+        dev_filetrans($1, tty_device_t, chr_file, "pty72")
+        dev_filetrans($1, tty_device_t, chr_file, "pty73")
+        dev_filetrans($1, tty_device_t, chr_file, "pty74")
+        dev_filetrans($1, tty_device_t, chr_file, "pty75")
+        dev_filetrans($1, tty_device_t, chr_file, "pty76")
+        dev_filetrans($1, tty_device_t, chr_file, "pty77")
+        dev_filetrans($1, tty_device_t, chr_file, "pty78")
+        dev_filetrans($1, tty_device_t, chr_file, "pty79")
+        dev_filetrans($1, tty_device_t, chr_file, "pty80")
+        dev_filetrans($1, tty_device_t, chr_file, "pty81")
+        dev_filetrans($1, tty_device_t, chr_file, "pty82")
+        dev_filetrans($1, tty_device_t, chr_file, "pty83")
+        dev_filetrans($1, tty_device_t, chr_file, "pty84")
+        dev_filetrans($1, tty_device_t, chr_file, "pty85")
+        dev_filetrans($1, tty_device_t, chr_file, "pty86")
+        dev_filetrans($1, tty_device_t, chr_file, "pty87")
+        dev_filetrans($1, tty_device_t, chr_file, "pty88")
+        dev_filetrans($1, tty_device_t, chr_file, "pty89")
+        dev_filetrans($1, tty_device_t, chr_file, "pty90")
+        dev_filetrans($1, tty_device_t, chr_file, "pty91")
+        dev_filetrans($1, tty_device_t, chr_file, "pty92")
+        dev_filetrans($1, tty_device_t, chr_file, "pty93")
+        dev_filetrans($1, tty_device_t, chr_file, "pty94")
+        dev_filetrans($1, tty_device_t, chr_file, "pty95")
+        dev_filetrans($1, tty_device_t, chr_file, "pty96")
+        dev_filetrans($1, tty_device_t, chr_file, "pty97")
+        dev_filetrans($1, tty_device_t, chr_file, "pty98")
+        dev_filetrans($1, tty_device_t, chr_file, "pty99")
+        dev_filetrans($1, tty_device_t, chr_file, "adb0")
+        dev_filetrans($1, tty_device_t, chr_file, "adb1")
+        dev_filetrans($1, tty_device_t, chr_file, "adb2")
+        dev_filetrans($1, tty_device_t, chr_file, "adb3")
+        dev_filetrans($1, tty_device_t, chr_file, "adb4")
+        dev_filetrans($1, tty_device_t, chr_file, "adb5")
+        dev_filetrans($1, tty_device_t, chr_file, "adb6")
+        dev_filetrans($1, tty_device_t, chr_file, "adb7")
+        dev_filetrans($1, tty_device_t, chr_file, "adb8")
+        dev_filetrans($1, tty_device_t, chr_file, "adb9")
+        dev_filetrans($1, tty_device_t, chr_file, "capi0")
+        dev_filetrans($1, tty_device_t, chr_file, "capi1")
+        dev_filetrans($1, tty_device_t, chr_file, "capi2")
+        dev_filetrans($1, tty_device_t, chr_file, "capi3")
+        dev_filetrans($1, tty_device_t, chr_file, "capi4")
+        dev_filetrans($1, tty_device_t, chr_file, "capi5")
+        dev_filetrans($1, tty_device_t, chr_file, "capi6")
+        dev_filetrans($1, tty_device_t, chr_file, "capi7")
+        dev_filetrans($1, tty_device_t, chr_file, "capi8")
+        dev_filetrans($1, tty_device_t, chr_file, "capi9")
+        dev_filetrans($1, console_device_t, chr_file, "console")
+        dev_filetrans($1, tty_device_t, chr_file, "cu0")
+        dev_filetrans($1, tty_device_t, chr_file, "cu1")
+        dev_filetrans($1, tty_device_t, chr_file, "cu2")
+        dev_filetrans($1, tty_device_t, chr_file, "cu3")
+        dev_filetrans($1, tty_device_t, chr_file, "cu4")
+        dev_filetrans($1, tty_device_t, chr_file, "cu5")
+        dev_filetrans($1, tty_device_t, chr_file, "cu6")
+        dev_filetrans($1, tty_device_t, chr_file, "cu7")
+        dev_filetrans($1, tty_device_t, chr_file, "cu8")
+        dev_filetrans($1, tty_device_t, chr_file, "cu9")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri0")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri1")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri2")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri3")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri4")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri5")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri6")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri7")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri8")
+        dev_filetrans($1, tty_device_t, chr_file, "dcbri9")
+        dev_filetrans($1, tty_device_t, chr_file, "vcsa")
+        dev_filetrans($1, tty_device_t, chr_file, "vcsb")
+        dev_filetrans($1, tty_device_t, chr_file, "vcsc")
+        dev_filetrans($1, tty_device_t, chr_file, "vcsd")
+        dev_filetrans($1, tty_device_t, chr_file, "vcse")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc0")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc1")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc2")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc3")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc4")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc5")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc6")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc7")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc8")
+        dev_filetrans($1, tty_device_t, chr_file, "hvc9")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi0")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi1")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi2")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi3")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi4")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi5")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi6")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi7")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi8")
+        dev_filetrans($1, tty_device_t, chr_file, "hvsi9")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm0")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm1")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm2")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm3")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm4")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm5")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm6")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm7")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm8")
+        dev_filetrans($1, tty_device_t, chr_file, "ircomm9")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn0")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn1")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn2")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn3")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn4")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn5")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn6")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn7")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn8")
+        dev_filetrans($1, tty_device_t, chr_file, "isdn9")
+        filetrans_pattern($1, devpts_t, ptmx_t, chr_file, "ptmx")
+        dev_filetrans($1, ptmx_t, chr_file, "ptmx")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm0")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm1")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm2")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm3")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm4")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm5")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm6")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm7")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm8")
+        dev_filetrans($1, tty_device_t, chr_file, "rfcomm9")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr0")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr1")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr2")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr3")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr4")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr5")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr6")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr7")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr8")
+        dev_filetrans($1, tty_device_t, chr_file, "slamr9")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM0")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM1")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM2")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM3")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM4")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM5")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM6")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM7")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM8")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyACM9")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS0")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS1")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS2")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS3")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS4")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS5")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS6")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS7")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS8")
+        dev_filetrans($1, tty_device_t, chr_file, "ttyS9")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG0")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG1")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG2")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG3")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG4")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG5")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG6")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG7")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG8")
+        dev_filetrans($1, tty_device_t, chr_file, "ttySG9")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB0")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB1")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB2")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB3")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB4")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB5")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB6")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB7")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB8")
+        dev_filetrans($1, usbtty_device_t, chr_file, "ttyUSB9")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p0")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p1")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p2")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p3")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p4")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p5")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p6")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p7")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p8")
+        dev_filetrans($1, virtio_device_t, chr_file, "vport0p9")
+        dev_filetrans($1, devpts_t, dir, "pts")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc0")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc1")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc2")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc3")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc4")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc5")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc6")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc7")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc8")
+        dev_filetrans($1, tty_device_t, chr_file, "xvc9")
+')
--- refpolicy.orig/policy/modules/roles/sysadm.te
+++ refpolicy/policy/modules/roles/sysadm.te
@@ -51,6 +51,8 @@ init_reload_generic_units(sysadm_t)
 userdom_manage_user_home_dirs(sysadm_t)
 userdom_home_filetrans_user_home_dir(sysadm_t)
 
+term_filetrans_all_named_dev(sysadm_t)
+
 ifdef(`init_systemd',`
 	init_get_all_units_status(sysadm_t)
 	init_start_all_units(sysadm_t)
--- refpolicy.orig/policy/modules/system/logging.te
+++ refpolicy/policy/modules/system/logging.te
@@ -499,6 +499,7 @@ fs_getattr_all_fs(syslogd_t)
 fs_search_auto_mountpoints(syslogd_t)
 fs_search_tmpfs(syslogd_t)
 
+mls_file_read_all_levels(syslogd_t)
 mls_file_write_all_levels(syslogd_t) # Need to be able to write to /var/run/ and /var/log directories
 mls_socket_write_all_levels(syslogd_t) # Neet to be able to sendto dgram
 mls_trusted_object(syslogd_t) # Other process need to have the right to connectto/sendto /dev/log
